add this back because redhatters like to make things
a pain for us even if it's zero effort for them not to

diff --git a/data/meson.build b/data/meson.build
index 7b8a0ea..4911b54 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -62,7 +62,7 @@
   build_conf = configuration_data()
   build_conf.set('libexecdir', libexecdir)
 
-  if libsystemd.found()
+  if libsystemd.found() or elogind.found()
     build_conf.set('bindir', bindir)
     build_conf.set('datadir', datadir)
     build_conf.set('localstatedir', localstatedir)
@@ -158,29 +158,6 @@
     endif
     build_conf.set('dynamic_options', '\n'.join(dynamic_options))
     build_conf.set('motd_dir', motd_dir)
-
-    # replace @dynamic_options@
-    configure_file(
-      input: 'fwupd.service.in',
-      output: 'fwupd.service',
-      configuration: build_conf,
-      install: true,
-      install_dir: systemdunitdir,
-    )
-
-    # for various plugins
-    install_data(['fwupd-i2c.conf'],
-      install_dir: systemd_modules_load_dir,
-    )
-
-    # for activation
-    configure_file(
-      input: 'fwupd.shutdown.in',
-      output: 'fwupd.shutdown',
-      configuration: build_conf,
-      install: true,
-      install_dir: systemd_shutdown_dir,
-    )
   else
     build_conf.set('systemd_service', '')
   endif
diff --git a/meson.build b/meson.build
index 7b8a0ea..4911b54 100644
--- a/meson.build
+++ b/meson.build
@@ -514,6 +514,11 @@ if libsystemd.found()
   endif
 endif
 
+elogind = dependency('systemd', 'libelogind', required: get_option('elogind'))
+if elogind.found()
+  conf.set('HAVE_LOGIND' , '1')
+endif
+
 supported_build = get_option('supported_build').disable_auto_if(not tag).allowed()
 if supported_build
     conf.set('SUPPORTED_BUILD', '1')
diff --git a/meson_options.txt b/meson_options.txt
index f880269..bd491cc 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -165,6 +165,10 @@ option('systemd',
   type: 'feature',
   description: 'systemd support',
 )
+option('elogind',
+  type: 'feature',
+  description: 'elogind support',
+)
 option('systemd_root_prefix',
   type: 'string',
   value: '',
diff --git a/plugins/logind/meson.build b/plugins/logind/meson.build
index bd66fabbc..7fe59f594 100644
--- a/plugins/logind/meson.build
+++ b/plugins/logind/meson.build
@@ -1,4 +1,4 @@
-libsystemd.found() or subdir_done()
+libsystemd.found() or elogind.found() or subdir_done()
 
 cargs = ['-DG_LOG_DOMAIN="FuPluginLogind"']
 plugins += {meson.current_source_dir().split('/')[-1]: true}
